packages/reports/addon/templates/components/report-view.hbs

```diff
@@ -1,45 +1,64 @@
-            <NaviIcon @icon="pencil" />
-          </div>
-        {{/animated-if}}
-        {{#animated-if (not this.hasRequestRun) use=this.fadeTransition}}
-          <div class="report-view__info-text">
-          <NaviIcon @icon="exclamation-circle" /> Run request to update {{this.visualizationTypeLabel}}</div>
-        {{/animated-if}}
-    </div>
-    <ReportVisualization
-      class="report-view__visualization"
-      @report={{@report}}
-      @response={{@response}}
-      @container={{this}}
-      @annotationData={{this.annotationData}}
-      @isEditing={{and this.isEditingVisualization @hasRequestRun}}
-      @onUpdateReport={{pipe
-        (route-action "onUpdateReport")
-        (route-action "validate" @report)
-        (route-action "runReport" @report)
-      }}
-    />
-    <MissingIntervalsWarning
-      @response={{this.response}}
-      @onDetailsToggle={{this.resizeVisualization}}
+  <div class="report-view__visualization-header">
+    <VisualizationToggle
+        @report={{@report}}
+        @validVisualizations={{this.validVisualizations}}
+        @onVisualizationTypeUpdate={{this.onVisualizationTypeUpdate}}
     />
-  {{/if}}
+      {{#animated-if (and this.hasRequestRun (not this.isEditingVisualization)) use=this.visFadeTransition}}
+        <div class="clickable report-view__visualization-edit-btn" role="button" {{on "click" this.toggleEditVisualization}}>
+          Edit {{this.visualizationTypeLabel}}
+          <NaviIcon @icon="pencil" />
+        </div>
+      {{/animated-if}}
+      {{#animated-if (not this.hasRequestRun) use=this.visFadeTransition}}
+        <div class="report-view__info-text">
+        <NaviIcon @icon="exclamation-circle" /> Run request to update {{this.visualizationTypeLabel}}</div>
+      {{/animated-if}}
+  </div>
+  <div class= "report-view__visualization-body">
+    {{#if (feature-flag "enableRequestPreview")}}
+      <div class="report-view__columns-toggle">
+        <button class="report-view__columns-button" type="button" aria-label="Toggle column drawer" {{on "click" (toggle "isColumnDrawerOpen" this)}}>
+          {{#animated-if this.isColumnDrawerOpen use=this.fadeTransition}}
+            <NaviIcon @icon="chevron-left" @classNames="report-view__columns-icon" />
+          {{else}}
+            <NaviIcon @icon="columns" @classNames="report-view__columns-icon" />
+          {{/animated-if}}
+        </button>
+      </div>
+    {{/if}}
+    {{#if hasNoData}}
+      <div class="report-view__visualization-no-results">
+        No results available.
+      </div>
+    {{else}}
+      <div class="report-view__visualization-container">
+        <ReportVisualization
+          class="report-view__visualization"
+          @report={{@report}}
+          @response={{@response}}
+          @container={{this}}
+          @annotationData={{this.annotationData}}
+          @isEditing={{and this.isEditingVisualization @hasRequestRun}}
+          @onUpdateReport={{pipe
```

Comment:
Does this need to be `pipe`? Can it be changed to `queue`?

Reply:
Probably